src/ripple/app/misc/impl/Transaction.cpp

```diff
@@ -169,11 +180,20 @@ Transaction::load(
         }
 
         convert(sociRawTxnBlob, rawTxn);
+        convert(sociRawMetaBlob, rawMeta);
     }
 
     try
     {
-        return Transaction::transactionFromSQL(ledgerSeq, status, rawTxn, app);
+        auto txn =
+            Transaction::transactionFromSQL(ledgerSeq, status, rawTxn, app);
+
+        std::uint32_t const inLedger =
+            rangeCheckedCast<std::uint32_t>(ledgerSeq.value_or(0));
```

Comment:
I'm not sure `ledgerSeq` should ever be unseated here. We should check if we ever store metadata without a ledger seq (I don't think we do, but it should be easy enough to audit). If we don't then it's an error to get metadata without a ledger seq, and we shouldn't fill in a dummy value here. I'd treat it as if the metadata was not in the database.